client/jetpack-cloud/sections/hooks/use-show-banners.ts

```diff
@@ -0,0 +1,39 @@
+import { ReactElement, useCallback } from 'react';
+import type { Preference } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+
+interface Banner {
+	component: () => ReactElement;
+	preference: Preference;
+	showDays?: number;
+	hideBanner?: boolean;
+}
+
+// Hook to show banner based on priority and preference(dismiss/view date)
+export const useShowBanners = ( banners: Array< Banner > ) => {
+	const showBanner = useCallback( ( banner: Banner ) => {
+		if ( ! banner.preference?.view_date ) {
+			return true;
+		}
+		const currentDate = new Date() as unknown as number;
+		const daysDifference = Math.floor(
+			( currentDate - Number( banner.preference.view_date ) ) / ( 1000 * 60 * 60 * 24 )
```

Comment:
It would be spectacular if we extracted `1000 * 60 * 60 * 24` into a separate constant so that instead of a "magic number" we have something like `ONE_DAY_IN_MILLISECONDS`.

Reply:
Agre. Updated here: https://github.com/Automattic/wp-calypso/pull/79510/commits/c05c1b689898087cbe6031189d7cf2bd8689e196